From: John Ferlito <johnf@inodes.org>
Date: Fri, 18 Feb 2011 00:01:49 +1100
Subject: [PATCH] Use self.ui rather than UIBase.getglobalui()

We have vonverted all places in folder/* to have self.ui available,
rather than having to use UIBase.getglobalui() all the
time. Unfortunately, we did not convert the users in folder/Base.py.
This patch does it belatedly. This fixes
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=613483
---
 offlineimap/folder/Base.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/offlineimap/folder/Base.py b/offlineimap/folder/Base.py
index 0524ec9..8e6a6b3 100644
--- a/offlineimap/folder/Base.py
+++ b/offlineimap/folder/Base.py
@@ -293,7 +293,7 @@ class BaseFolder:
         except (KeyboardInterrupt):
             raise
         except:
-            UIBase.getglobalui().warn("ERROR attempting to copy message " + str(uid) \
+            self.ui.warn("ERROR attempting to copy message " + str(uid) \
                  + " for account " + self.getaccountname() + ":" + traceback.format_exc())
         
 
@@ -400,7 +400,7 @@ class BaseFolder:
         except (KeyboardInterrupt):
             raise
         except:
-            UIBase.getglobalui().warn("ERROR attempting to handle negative uids " \
+            self.ui.warn("ERROR attempting to handle negative uids " \
                 + "for account " + self.getaccountname() + ":" + traceback.format_exc())
 
         #all threads launched here are in try / except clauses when they copy anyway...
@@ -411,7 +411,7 @@ class BaseFolder:
         except (KeyboardInterrupt):
             raise
         except:
-            UIBase.getglobalui().warn("ERROR attempting to delete messages " \
+            self.ui.warn("ERROR attempting to delete messages " \
                 + "for account " + self.getaccountname() + ":" + traceback.format_exc())
 
         # Now, the message lists should be identical wrt the uids present.
@@ -423,7 +423,7 @@ class BaseFolder:
         except (KeyboardInterrupt):
             raise
         except:
-            UIBase.getglobalui().warn("ERROR attempting to sync flags " \
+            self.ui.warn("ERROR attempting to sync flags " \
                 + "for account " + self.getaccountname() + ":" + traceback.format_exc())
         
             
-- 
